scripts/zones/Uleguerand_Range/mobs/Mountain_Worm_NM.lua

```diff
@@ -0,0 +1,24 @@
+-----------------------------------
+-- Area: Uleguerand Range
+--  Mob: Mountain Worm NM
+--  https://www.bg-wiki.com/ffxi/Mountain_Worm_(NM)
+-- TODO allow deaggro based on distance (core CMobEntity::CanDeaggro() forces NM and Battlefield mobs to never stop chasing)
```

Comment:
Its technically not the distance (I think) but just how long it goes without being able to target/attack you in any way. which of course would mean you are out of casting range.